### ISSUE TITLE
Upgrade react-native-reanimated 2.14.1 to 3.3.0 Failed in iOS into Existing App project.

### ISSUE THREAD

**Issue by Nikunjp1904** [software-mansion/react-native-reanimated#4614]
### Description

After upgrade react-native from "react-native": "0.68.1" to "react-native": "0.72.0", then update the react-native-reanimated from 2.14.1 to 3.3.0. in IOS getting issue as attached screenshot.
![Screenshot 2023-06-22 at 9 48 40 PM](https://github.com/software-mansion/react-native-reanimated/assets/40294521/ac6dd8ed-69ef-4133-84b0-0192a04015f4)

Error is like
Error: [Reanimated] The native part of Reanimated doesn't seem to be initialized. This could be caused by
  - not rebuilding the app after installing or upgrading Reanimated
  
Already added babel config like
```
const plugins = [
  ........
    [
      'react-native-reanimated/plugin',
      {
        relativeSourceLocation: true,
      },
    ],
  ]
```

### Steps to reproduce

1. upgrade react-native from "react-native": "0.68.1" to "react-native": "0.72.0"
2.  update the react-native-reanimated from 2.14.1 to 3.3.0.
3. pod update and then run in iOS getting above error

### Snack or a link to a repository

No snack

### Reanimated version

3.3.0

### React Native version

0.72.0

### Platforms

iOS

### JavaScript runtime

None

### Workflow

None

### Architecture

None

### Build type

Debug mode

### Device

iOS simulator

### Acknowledgements

Yes

**Comment by stbdang** [comment]
So I did some digging around and here is my finding.

1. Exception '-[RCTUIManager setFlushUiOperations:]: - The code is trying to set RCTUIManager.flushUiOperations property and ObjC runtime is complaining that there's no such selector
2. Debugging into the code reveals that object is not an instance of RCTUIManager but rather the superclass (RCTUIManager)
3. During REAInitializer, it seems to register this new RCTUIManager to replace vanilla RCTUIManager
4. However, REAInitializer is never called on my app (RN enabled via https://reactnative.dev/docs/integration-with-existing-apps)
5. Replicating it on a new RN app shows that REAInitializer is called.
6. Looks like RCTCxxBridge only calls jsExecutorFactoryForBridge (which calls REAInitializer) if the delegate conforms to RCTCxxBridgeDelegate.
7. Vanilla RN app implements RCTAppDelegate, which conforms to RCTCxxBridgeDelegate
8. Migrated Native apps don't use RCTAppDelegate by default. Instead, it uses its own implementation.
9. Switching my AppDelegate to conform to RCTAppDelegate would probably get REAInitializer to be called. I'm not sure what the implication of that is, though.

Will give it a try and update this thread.

**Comment by stbdang** [comment]
I just learned that RCTAppDelegate also acts as a bridge delegate... this will be fun.

**Comment by stbdang** [comment]
With my app, we are manually creating an instance of RCTBridge and holding its reference to host multiple RN components. I think we could get it to work by calling REAInitializer when we create RCTBridge. I will try it next time when I get a chance - don't need reanimated just yet.

**Comment by felipegmg** [comment]
same issue here!

**Comment by piaskowyk** [comment]
We got rid of overwriting UIManager here - https://github.com/software-mansion/react-native-reanimated/pull/4619 It will be available in 3.4.0 - the release should be soon.

**Comment by rohit-jindal018** [comment]
Until react-native-reanimated 3.4.0 is released, one can directly include main branch of the package

react-native-reanimated: "https://github.com/software-mansion/react-native-reanimated.git#main"
This should fix the error.
